.github/workflows/deploy.yml

```diff
@@ -11,26 +11,21 @@ permissions:
 
 jobs:
   blackduck:
-    name: Blackduck
-    runs-on: pss-cyberpunk
-    permissions:
-      actions: read
-      contents: read
-      security-events: write
-
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
-
-      - uses: actions/setup-java@v3
-        with:
-          distribution: 'adopt'
-          java-version: '17'
-
-      - name: Software Composition Analysis with Black Duck
-        uses: blackducksoftware/github-action@9ea442b34409737f64743781e9adc71fd8e17d38
-        with:
-          args: '--blackduck.url="${{ secrets.BLACKDUCK_URL }}" --blackduck.api.token="${{ secrets.BLACKDUCK_TOKEN }}"'
+      - name: Black Duck Scan
+        env:
+          BLACKDUCK_URL: ${{ secrets.BLACKDUCK_URL }}
+          BLACKDUCK_API_TOKEN: ${{ secrets.BLACKDUCK_TOKEN }}
+          DETECT_PROJECT_NAME: ${{ github.event.repository.name }}
+          DETECT_PROJECT_VERSION_NAME: ${{ github.ref_name }}
+          DETECT_CODE_LOCATION_NAME: ${{ github.event.repository.name }}-${{ github.ref_name }}
+          DETECT_EXCLUDED_DETECTOR_TYPES: GIT
+        run: |
+          curl -fLsS -o $RUNNER_TEMP/detect8.sh https://detect.synopsys.com/detect8.sh && chmod +x $RUNNER_TEMP/detect8.sh
+          $RUNNER_TEMP/detect8.sh
```

Comment:
```suggestion
          curl -fLsS -o - https://detect.synopsys.com/detect8.sh | sh
```

What do you think?

Reply:
I think it's good but I want curl to exit the pipe if it fails to download the script. Is there a way to do that? I don't want to pipe garbage.

Reply:
Well I just realized that the script is specifically a `#!/bin/bash` script, so instead of `| sh` should probably be `| bash` or `| /bin/bash`

Reply:
Makes sense. What about if curl fails to download it though?

Reply:
Not sure if anything bad could happen, if the curl command gives no output `bash` will be invoked with no input, effectively exiting immediately.

Reply:
Yes, I meant if curl produced something bad. But that goes to stderr, it seems, so that's okay. But if bash exits with 0, the behavior is different from what it is now where the job will fail if the file is not downloaded (either running it fails or chmodding it fails).

Reply:
I see your concern, sounds valid to me and I did not consider it. I think you can decide if you want to keep it as is, or set pipefail and use this one. (or if there's another, better way) Resolve when you're happy!

Reply:
pipefail is apparently set by github, so the command is fine as you suggest. Changed it (and tested in check).

Reply:
Very cool!